projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,140 +1,345 @@
+        describe('without data', () => {
+            it('should be pristine, untouched & valid', async () => {
+                await page.setContent(`<dot-textfield></dot-textfield>`);
+                element = await page.find('dot-textfield');
+                expect(element).toHaveClasses(dotTestUtil.class.empty);
+            });
+        });
     });
 
-    it('should clear value, set pristine and untouched  when input set reset', async () => {
-        await element.callMethod('reset');
-        await page.waitForChanges();
+    describe('@Props', () => {
+        beforeEach(async () => {
+            page = await newE2EPage({
+                html: `<dot-textfield></dot-textfield>`
+            });
 
-        expect(element.classList.contains('dot-pristine')).toBe(true);
-        expect(element.classList.contains('dot-untouched')).toBe(true);
-        expect(element.classList.contains('dot-invalid')).toBe(true);
-        expect(await input.getProperty('value')).toBe('');
-    });
+            element = await page.find('dot-textfield');
+            input = await page.find('input');
+        });
 
-    it('should mark as disabled when prop is present', async () => {
-        element.setProperty('disabled', true);
-        await page.waitForChanges();
-        expect(await input.getProperty('disabled')).toBe(true);
-    });
+        describe('value', () => {
+            it('should set value correctly', async () => {
+                await input.press('a');
+                await page.waitForChanges();
+                expect(await input.getProperty('value')).toBe('a');
+            });
+            it('should render and not break when is a unexpected value', async () => {
+                element.setProperty('value', { test: true });
+                await page.waitForChanges();
+                expect(await input.getProperty('value')).toBe('[object Object]');
+            });
+        });
 
-    it('should mark as required when prop is present', async () => {
-        expect(await input.getProperty('required')).toBe(true);
-    });
+        describe('name', () => {
+            it('should render with valid name', async () => {
+                element.setProperty('name', 'text01');
+                await page.waitForChanges();
 
-    it('should set the default value of regexCheck when the Regular Expression is not valid', async () => {
-        element.setProperty('regexCheck', '[^(<[.\\n]+>)]*l');
-        await page.waitForChanges();
-        expect(await input.getProperty('regexCheck')).toBeUndefined();
-    });
+                expect(input.getAttribute('id')).toBe('dot-text01');
+            });
+
+            it('should set name prop in dot-label', async () => {
+                element.setProperty('name', 'text01');
+                await page.waitForChanges();
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(label.getAttribute('name')).toBe('text01');
```

Comment:
You should also tests the `id` of the `<input>` because its depends of the name, see [here](https://github.com/dotCMS/core-web/blob/af08b7a6548aa70a38660e113b5f8e1a6a59ca78/projects/dotcms-field-elements/src/components/dot-textfield/dot-textfield.tsx#L110).

And the same goes with all the other prop make sure you test all that the prop affects.

Reply:
is tested in the previous test and the last one of this prop.